src/main/java/de/tum/in/www1/artemis/service/user/UserService.java

```diff
@@ -376,6 +376,37 @@ public Optional<User> createUserFromLdap(String registrationNumber) {
         return Optional.empty();
     }
 
+    /**
+     * Searches the (optional) LDAP service for a user with the give registration number (= Matrikelnummer) and updates the Artemis user with
+     * the information provided by the LDAP server.
+     *
+     * @param registrationNumber the matriculation number of the student
+     * @return the user with the given registration number or null if no user with the given registration number was found
+     */
+    public Optional<User> updateUserFromLdap(String registrationNumber) {
```

Comment:
This method is mostly duplicated from `createUserFromLdap`
I don't think this is a good idea. I suggest we combine both into one method.
Please notice that `createUserFromLdap` already handles the case that the user exists in the Artemis database and then updates the user accordingly

Reply:
When following the changes below, this method might not even be needed and could be removed